modules/bioinspired/include/opencv2/bioinspired/retina.hpp

```diff
@@ -446,9 +447,9 @@ the log scale that is applied
  */
 CV_EXPORTS_W Ptr<Retina> createRetina(Size inputSize, const bool colorMode, int colorSamplingMethod=RETINA_COLOR_BAYER, const bool useRetinaLogSampling=false, const float reductionFactor=1.0f, const float samplingStrenght=10.0f);
 
-#ifdef HAVE_OPENCV_OCL
-Ptr<Retina> createRetina_OCL(Size inputSize);
-Ptr<Retina> createRetina_OCL(Size inputSize, const bool colorMode, int colorSamplingMethod=RETINA_COLOR_BAYER, const bool useRetinaLogSampling=false, const float reductionFactor=1.0f, const float samplingStrenght=10.0f);
+#ifdef HAVE_OPENCL
+CV_EXPORTS_W Ptr<Retina> createRetina_OCL(Size inputSize);
+CV_EXPORTS_W Ptr<Retina> createRetina_OCL(Size inputSize, const bool colorMode, int colorSamplingMethod = RETINA_COLOR_BAYER, const bool useRetinaLogSampling = false, const double reductionFactor = 1.0f, const double samplingStrenght = 10.0f);
 #endif
```

Comment:
Problem with binding is here. `#ifdef` should be removed from public headers (with `#include "cvconfig.h"`)

BTW, 3.x OpenCL API is "transparent" (T-API). It means that the same user code should work on platforms with and without OpenCL support. So public API should not contain separate "_OCL" functions.